File: Scripts/results/generated_code_results/Flutter/Education/WiseOnlineCourses/MyListScreen.dart
import 'package:flutter/material.dart';
class MyListScreen extends StatefulWidget {
@override
_MyListScreenState createState() => _MyListScreenState();
}
class _MyListScreenState extends State<MyListScreen> with SingleTickerProviderStateMixin {
late TabController _tabController;
@override
void initState() {
super.initState();
_tabController = TabController(length: 2, vsync: this);
}
@override
void dispose() {
_tabController.dispose();
super.dispose();
}
@override
Widget build(BuildContext context) {
return Scaffold(
backgroundColor: Color(0xFF1E1E1E),
appBar: AppBar(
backgroundColor: Colors.transparent,
elevation: 0,
title: TabBar(
controller: _tabController,
tabs: [
Tab(text: 'Home'),
Tab(text: 'My List'),
],
indicatorColor: Colors.yellow,
labelColor: Colors.white,
unselectedLabelColor: Colors.grey,
),
actions: [
Padding(
padding: const EdgeInsets.all(8.0),
child: CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
),
),
],
),
body: ListView(
padding: EdgeInsets.all(16),
children: [
Text(
'Instructors',
style: TextStyle(color: Colors.white, fontSize: 24, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
_buildInstructorItem(),
SizedBox(height: 24),
Text(
'Lessons',
style: TextStyle(color: Colors.white, fontSize: 24, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
_buildLessonItem(
'Gift yourself this UI Kit and enjoy!',
'John Wiseberg',
'08:12',
),
SizedBox(height: 16),
_buildLessonItem(
'Make your Mess Your Message',
'Mendy Santiago',
'12:38',
),
],
),
bottomNavigationBar: BottomNavigationBar(
backgroundColor: Color(0xFF1E1E1E),
selectedItemColor: Colors.white,
unselectedItemColor: Colors.grey,
currentIndex: 0,
items: [
BottomNavigationBarItem(
icon: Icon(Icons.home),
label: 'Home',
),
BottomNavigationBarItem(
icon: Icon(Icons.search),
label: 'Search',
),
BottomNavigationBarItem(
icon: Icon(Icons.file_download),
label: 'Downloads',
),
],
),
);
}
Widget _buildInstructorItem() {
return GestureDetector(
onTap: () => Navigator.pushNamed(context, 'CourseDetail'),
child: Container(
padding: EdgeInsets.all(12),
decoration: BoxDecoration(
color: Color(0xFF2A2A2A),
borderRadius: BorderRadius.circular(8),
),
child: Row(
children: [
ClipRRect(
borderRadius: BorderRadius.circular(8),
child: Image.asset(
'assets/images/template.png',
width: 60,
height: 60,
fit: BoxFit.cover,
),
),
SizedBox(width: 16),
Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
'Boniface Esanji',
style: TextStyle(color: Colors.white, fontSize: 18, fontWeight: FontWeight.bold),
),
Text(
'Teaches Guitar',
style: TextStyle(color: Colors.grey, fontSize: 14),
),
],
),
],
),
),
);
}
Widget _buildLessonItem(String title, String instructor, String duration) {
return GestureDetector(
onTap: () => Navigator.pushNamed(context, 'CourseDetail'),
child: Container(
padding: EdgeInsets.all(12),
decoration: BoxDecoration(
color: Color(0xFF2A2A2A),
borderRadius: BorderRadius.circular(8),
),
child: Row(
children: [
Stack(
children: [
ClipRRect(
borderRadius: BorderRadius.circular(8),
child: Image.asset(
'assets/images/template.png',
width: 120,
height: 80,
fit: BoxFit.cover,
),
),
Positioned(
left: 4,
bottom: 4,
child: Container(
padding: EdgeInsets.symmetric(horizontal: 4, vertical: 2),
decoration: BoxDecoration(
color: Colors.black.withOpacity(0.7),
borderRadius: BorderRadius.circular(4),
),
child: Text(
duration,
style: TextStyle(color: Colors.white, fontSize: 12),
),
),
),
],
),
SizedBox(width: 16),
Expanded(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
title,
style: TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold),
),
SizedBox(height: 4),
Text(
instructor,
style: TextStyle(color: Colors.grey, fontSize: 14),
),
],
),
),
],
),
),
);
}
}

File: Scripts/results/generated_code_results/baseline_claude/Education/WiseOnlineCourses/MyListScreen.js
import React from 'react';

import { View, Text, Image, StyleSheet, ScrollView, TouchableOpacity } from 'react-native';

import { useNavigation } from '@react-navigation/native';



const InstructorCard = ({ image, name, description }) => (

  <View style={styles.instructorCard}>

    <Image source={image} style={styles.instructorImage} />

    <View style={styles.instructorInfo}>

      <Text style={styles.instructorName}>{name}</Text>

      <Text style={styles.instructorDescription}>{description}</Text>

    </View>

  </View>

);



const LessonCard = ({ image, title, author, duration }) => (

  <View style={styles.lessonCard}>

    <Image source={image} style={styles.lessonImage} />

    <View style={styles.lessonInfo}>

      <Text style={styles.lessonTitle}>{title}</Text>

      <Text style={styles.lessonAuthor}>{author}</Text>

    </View>

    <Text style={styles.lessonDuration}>{duration}</Text>

  </View>

);



const MyListScreen = () => {

  const navigation = useNavigation();



  const instructors = [

    { id: '1', image: require('../assets/snack-icon.png'), name: 'Boniface Esanji', description: 'Teaches Guitar' },

  ];



  const lessons = [

    { id: '1', image: require('../assets/snack-icon.png'), title: 'Gift yourself this UI Kit and enjoy!', author: 'John Wiseberg', duration: '08:12' },

    { id: '2', image: require('../assets/snack-icon.png'), title: 'Make your Mess Your Message', author: 'Mendy Santiago', duration: '12:38' },

  ];



  return (

    <View style={styles.container}>

      <View style={styles.header}>

        <TouchableOpacity onPress={() => navigation.navigate('Home')}>

          <Text style={styles.tabText}>Home</Text>

        </TouchableOpacity>

        <View style={styles.activeTabContainer}>

          <Text style={styles.activeTabText}>My List</Text>

          <View style={styles.activeTabIndicator} />

        </View>

        <Image source={require('../assets/snack-icon.png')} style={styles.profileImage} />

      </View>



      <ScrollView style={styles.content}>

        <Text style={styles.sectionTitle}>Instructors</Text>

        {instructors.map((instructor) => (

          <InstructorCard key={instructor.id} {...instructor} />

        ))}



        <Text style={styles.sectionTitle}>Lessons</Text>

        {lessons.map((lesson) => (

          <LessonCard key={lesson.id} {...lesson} />

        ))}

      </ScrollView>



      <View style={styles.tabBar}>

        <TouchableOpacity>

          <Image source={require('../assets/snack-icon.png')} style={[styles.tabIcon, styles.activeTabIcon]} />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Search')}>

          <Image source={require('../assets/snack-icon.png')} style={styles.tabIcon} />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Downloads')}>

          <Image source={require('../assets/snack-icon.png')} style={styles.tabIcon} />

        </TouchableOpacity>

      </View>

    </View>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#1E1E1E',

  },

  header: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    alignItems: 'center',

    padding: 20,

  },

  tabText: {

    color: '#999999',

    fontSize: 18,

  },

  activeTabContainer: {

    alignItems: 'center',

  },

  activeTabText: {

    color: 'white',

    fontSize: 18,

    fontWeight: 'bold',

  },

  activeTabIndicator: {

    width: 30,

    height: 2,

    backgroundColor: '#FFD700',

    marginTop: 5,

  },

  profileImage: {

    width: 40,

    height: 40,

    borderRadius: 20,

  },

  content: {

    flex: 1,

    paddingHorizontal: 20,

  },

  sectionTitle: {

    color: 'white',

    fontSize: 22,

    fontWeight: 'bold',

    marginVertical: 15,

  },

  instructorCard: {

    flexDirection: 'row',

    alignItems: 'center',

    backgroundColor: '#2C2C2C',

    borderRadius: 10,

    padding: 15,

    marginBottom: 15,

  },

  instructorImage: {

    width: 60,

    height: 60,

    borderRadius: 30,

  },

  instructorInfo: {

    marginLeft: 15,

  },

  instructorName: {

    color: 'white',

    fontSize: 18,

    fontWeight: 'bold',

  },

  instructorDescription: {

    color: '#999999',

    fontSize: 14,

  },

  lessonCard: {

    flexDirection: 'row',

    alignItems: 'center',

    backgroundColor: '#2C2C2C',

    borderRadius: 10,

    padding: 15,

    marginBottom: 15,

  },

  lessonImage: {

    width: 80,

    height: 60,

    borderRadius: 5,

  },

  lessonInfo: {

    flex: 1,

    marginLeft: 15,

  },

  lessonTitle: {

    color: 'white',

    fontSize: 16,

    fontWeight: 'bold',

  },

  lessonAuthor: {

    color: '#999999',

    fontSize: 14,

  },

  lessonDuration: {

    color: '#999999',

    fontSize: 12,

  },

  tabBar: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    alignItems: 'center',

    height: 60,

    backgroundColor: '#2C2C2C',

  },

  tabIcon: {

    width: 24,

    height: 24,

    tintColor: '#999999',

  },

  activeTabIcon: {

    tintColor: 'white',

  },

});



export default MyListScreen;